server/src/main.ts

```diff
@@ -307,13 +308,17 @@ deviceSocket.on("connection", (ws, req) => {
     checkers.checkDeviceMessageLimit();
     /* eslint-disable-next-line @typescript-eslint/no-base-to-string */
     let messageStr = message.toString();
+    // There can be different variants of a message depending on the context of
+    // the receiver.
+    const forwardedMessage = createForwardedMessage();
```

Comment:
It looks like an unnecessary abstraction to me

Reply:
I removed it